test/k8s/egress.go

```diff
@@ -353,8 +362,18 @@ var _ = SkipDescribeIf(func() bool {
 
 				BeforeAll(func() {
 					policyYAML = applyEgressPolicy("egress-gateway-policy-excl-cidr.yaml")
-					kubectl.WaitForEgressPolicyEntry(k8s1IP, outsideIP)
-					kubectl.WaitForEgressPolicyEntry(k8s2IP, outsideIP)
+
+					// Wait for 4 entries:
+					// - 1 policies with:
+					//   - 2 matching endpoints
+					//   - 1 destination CIDR
+					//   - 1 excluded CIDR
+
+					err := kubectl.WaitForEgressPolicyEntries(helpers.K8s1, 4)
+					Expect(err).Should(BeNil(), "Failed waiting for egress policy map entries")
+
+					err = kubectl.WaitForEgressPolicyEntries(helpers.K8s2, 4)
+					Expect(err).Should(BeNil(), "Failed waiting for egress policy map entries")
```

Comment:
Should we stick some `WaitForEgressPolicyEntries(helpers.K8sX, 0)` into the `AfterAll()` calls as well, after removing the policy? To avoid overlap between old and new entries?